core/ipc/rpc/kafka/src/main/java/org/opennms/core/ipc/rpc/kafka/KafkaRpcClientFactory.java

```diff
@@ -127,14 +127,18 @@
     private String location;
     private KafkaProducer<String, byte[]> producer;
     private final Properties kafkaConfig = new Properties();
-    private final ThreadFactory threadFactory = new ThreadFactoryBuilder()
+    private final ThreadFactory consumerThreadFactory = new ThreadFactoryBuilder()
             .setNameFormat("rpc-client-kafka-consumer-%d")
             .build();
+    private final ThreadFactory responseHandlerThreadFactory = new ThreadFactoryBuilder()
+            .setNameFormat("rpc-client-response-handler-%d")
+            .build();
     private final ThreadFactory timerThreadFactory = new ThreadFactoryBuilder()
             .setNameFormat("rpc-client-timeout-tracker-%d")
             .build();
-    private final ExecutorService executor = Executors.newSingleThreadExecutor(threadFactory);
+    private final ExecutorService kafkaConsumerExecutor = Executors.newCachedThreadPool(consumerThreadFactory);
```

Comment:
Good point

Reply:
Yeah, I was doing that to start new thread when there is small chance of thread shutting down and `closed.set(true)`.  Since we are removing restarting thread logic, this is no longer needed